Auth/Auth.Platform.iOS/AuthService.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System;
+using AeroGear.Mobile.Auth.Config;
+using AeroGear.Mobile.Core;
+using AeroGear.Mobile.Core.Configuration;
+using static AeroGear.Mobile.Core.Utils.SanityCheck;
+
+namespace AeroGear.Mobile.Auth
+{
+    public class AuthService : IAuthService
+    {
+        private readonly KeycloakConfig keycloakConfig;
+        private readonly ServiceConfiguration serviceConfiguration;
+
+        private AuthService()
+        {
+            this.serviceConfiguration = NonNull(MobileCore.Instance.GetServiceConfiguration(Type), "configuration");
+            this.keycloakConfig = new KeycloakConfig(serviceConfiguration);
+        }
+
+        private AuthService(ServiceConfiguration configuration)
+        {
+            this.serviceConfiguration = NonNull(configuration, "configuration");
+            this.keycloakConfig = new KeycloakConfig(serviceConfiguration);
+        }
+
+        public string Type => "keycloak";
+
+        public bool RequiresConfiguration => true;
+
+        public void Configure(MobileCore core, ServiceConfiguration serviceConfiguration)
+        {
+        }
+
+        public void Destroy()
+        {
+        }
+
+        public static IAuthService InitializeService(ServiceConfiguration configuration)
+        {
+            return MobileCore.Instance.RegisterService<IAuthService>(new AuthService(configuration));
+        }
+
+        public static IAuthService InitializeService()
```

Comment:
do we need the 2 different initialise methods?

Reply:
@wei-lee Currently we have a way to override the configuration with `MobileCore`
* https://github.com/aerogear/aerogear-xamarin-sdk/blob/master/Core/Core/MobileCore.cs#L162]
* https://github.com/aerogear/aerogear-xamarin-sdk/blob/master/Core/Core/MobileCore.cs#L170

The second allows to pass a configuration. If we want to lose one of the options, than we can remove one of the initialise methods.